### PR TITLE
docs: #1308 update for alter partitions with kafka api

### LINKED ISSUE #1308: minor updates to alter partition
### Describe the Issue

missed this comment for alter partitions with kafka api: 
Include the flag `--preserve-throttles` to avoid errors when verifying or canceling a partition reassignment.

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/cluster-maintenance/cluster-balancing/

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1308 

preview differences between Kafka and Redpanda alter partitions: https://deploy-preview-1309--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/cluster-balancing/#differences-in-partition-balancing-between-redpanda-and-kafka

preview changing leadership with Admin API: https://deploy-preview-1309--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/cluster-balancing/#partition-leadership-balancing

### CODE REVIEW

**Comment by micheleRP** [comment]
Hi @mattschumpert  and @NyaliaLui. I was rushing to get out last week and missed updating this one comment. Is there anything else for this feature that needs updating for 23.1?

**Comment by micheleRP** [comment]
adding @daisukebe to review changing leadership with Admin API: https://deploy-preview-1309--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/cluster-balancing/#partition-leadership-balancing

File: docs/manage/cluster-maintenance/cluster-balancing.mdx [hunk 1]
```diff
@@ -13,9 +13,21 @@ With an Enterprise license, you can additionally enable *Continuous Data Balanci
...
 
-## Leadership balancing
+## Partition leadership balancing
 
-Automatic leadership balancing improves cluster performance by transferring leadership for a broker's partitions to other replicas. Leadership balancing is enabled by default with the `enable_leader_balancer` property. Automatic leadership balancing changes where data is read to and written to first. It doesn't move any data. For more information, see [Partition leadership elections](../../../get-started/architecture/#partition-leadership-elections).
+Automatic partition leadership balancing improves cluster performance by transferring leadership for a broker's partitions to other replicas. Leadership balancing is enabled by default with the `enable_leader_balancer` property. Automatic leadership balancing changes where data is read to and written to first. It doesn't move any data. For more information, see [Partition leadership elections](../../../get-started/architecture/#partition-leadership-elections).
+
+To manually change leadership, use the Admin API:
```
Comment (daisukebe) [inline docs/manage/cluster-maintenance/cluster-balancing.mdx:20]: I'm not sure how much we want to cover the k8s deployment scenario in this section though, let me commend one thing.

The `transfer_leadership` request has to be handled by the current leader partition broker. In a typical package based installations, redirection works behind the scenes, hence not a problem. However in a k8s environment where redirection doesn't typically work, we should encourage users to run the command on the current partition leader pod.

File: docs/manage/cluster-maintenance/cluster-balancing.mdx [hunk 2]
```diff
@@ -13,9 +13,21 @@ With an Enterprise license, you can additionally enable *Continuous Data Balanci
...
+
+```bash
+curl -X POST http://<broker_address>:9644/v1/partitions/kafka/<topic>/<partition>/transfer_leadership?target=<destination-broker-id>
+```
+
+For example, to change leadership for partition 0 to node 2:
+
+```bash
+curl -X POST "http://192.168.0.5:9644/v1/partitions/kafka/test2_replica3_parition6/0/transfer_leadership?target=2"
```
Comment (NyaliaLui) [inline docs/manage/cluster-maintenance/cluster-balancing.mdx:29]:
```suggestion
curl -X POST "http://localhost:9644/v1/partitions/kafka/test_topic/0/transfer_leadership?target=2"
```

**Review by NyaliaLui — COMMENTED** [review]
Just a couple small changes to the examples

File: docs/manage/cluster-maintenance/cluster-balancing.mdx (hunk 1 above)
Comment (micheleRP) [reply docs/manage/cluster-maintenance/cluster-balancing.mdx:20]: Thanks, Daisuke! I added this note: 
In Kubernetes, run the `transfer_leadership` request on the current partition leader Pod.

**Review by NyaliaLui — APPROVED** [review]

File: docs/manage/cluster-maintenance/cluster-balancing.mdx [hunk 3]
```diff
@@ -13,9 +13,25 @@ With an Enterprise license, you can additionally enable *Continuous Data Balanci
...
 
-## Leadership balancing
+## Partition leadership balancing
 
-Automatic leadership balancing improves cluster performance by transferring leadership for a broker's partitions to other replicas. Leadership balancing is enabled by default with the `enable_leader_balancer` property. Automatic leadership balancing changes where data is read to and written to first. It doesn't move any data. For more information, see [Partition leadership elections](../../../get-started/architecture/#partition-leadership-elections).
+Automatic partition leadership balancing improves cluster performance by transferring leadership for a broker's partitions to other replicas. Leadership balancing is enabled by default with the `enable_leader_balancer` property. Automatic leadership balancing changes where data is read to and written to first. It doesn't move any data. For more information, see [Partition leadership elections](../../../get-started/architecture/#partition-leadership-elections).
```
Comment (JakeSCahill) [inline docs/manage/cluster-maintenance/cluster-balancing.mdx:18]:
```suggestion
Automatic partition leadership balancing improves cluster performance by transferring the leadership of a broker's partitions to other replicas. Leadership balancing is enabled by default with the `enable_leader_balancer` property. Automatic leadership balancing changes where data is read from and written to first. It doesn't move any data. For more information, see [Partition leadership elections](../../../get-started/architecture/#partition-leadership-elections).
```

File: docs/manage/cluster-maintenance/cluster-balancing.mdx [hunk 4]
```diff
@@ -13,9 +13,25 @@ With an Enterprise license, you can additionally enable *Continuous Data Balanci
...
+```
+
+For example, to change leadership to broker 2 for partition 0 on topic `foo`:
+
+```bash
+curl -X POST "http://localhost:9644/v1/partitions/kafka/foo/0/transfer_leadership?target=2"
+```
+
+:::note
+In Kubernetes, run the `transfer_leadership` request on the current partition leader Pod.
```
Comment (JakeSCahill) [inline docs/manage/cluster-maintenance/cluster-balancing.mdx:33]:
```suggestion
In Kubernetes, run the `transfer_leadership` request on the Pod that is running the current partition leader.
```

**Review by JakeSCahill — APPROVED** [review]
LGTM - See comments for some minor suggestions.

**Pull request merged**